stats/condition_interceptor.go

```diff
@@ -0,0 +1,167 @@
+package stats
+
+import (
+	"errors"
+	"log"
+
+	"github.com/intervention-engine/fhir/models"
+	"github.com/intervention-engine/fhir/server"
+)
+
+// ConditionStatsCreateInterceptor intercepts any new condition resources added to the database
+// and updates the Synthetic Mass condition statistics based on the condition's patient's address.
+type ConditionStatsCreateInterceptor struct {
+	PgDataAccess    StatsDataAccess
+	MongoDataAccess server.DataAccessLayer
+}
+
+// Before is unused
+func (s *ConditionStatsCreateInterceptor) Before(resource interface{}) {}
+
+// After increments disease statistics for given after a new condition resource is created.
+func (s *ConditionStatsCreateInterceptor) After(resource interface{}) {
+	condition, ok := resource.(*models.Condition)
+
+	if ok {
+
+		var err error
+
+		if condition.Subject == nil {
+			log.Printf("ConditionStatsCreateInterceptor: Condition %s has no subject\n", condition.Id)
+			return
+		}
+
+		result, err := s.MongoDataAccess.Get(condition.Subject.ReferencedID, "Patient")
+
+		if err != nil {
+			log.Printf("ConditionStatsCreateInterceptor: Failed to get patient for condition %s\n", condition.Id)
+			return
+		}
+
+		if !conditionIsAbated(condition) {
+			patient := result.(*models.Patient)
+			err = s.PgDataAccess.AddConditionStat(patient, condition)
+		}
+
+		if err != nil {
+			log.Printf("ConditionStatsCreateInterceptor: %s\n", err.Error())
+		}
+	}
+}
+
+// OnError is unused
+func (s *ConditionStatsCreateInterceptor) OnError(err error, resource interface{}) {}
+
+// ConditionStatsUpdateInterceptor intercepts any updates to condition resources in the database
+// and updates the Synthetic Mass condition statistics based on the condition's patient's address.
+type ConditionStatsUpdateInterceptor struct {
+	PgDataAccess    StatsDataAccess
+	MongoDataAccess server.DataAccessLayer
+	// The state of the condition before the database update, for comparison after the database update
+	conditionBefore *models.Condition
+	// Tracks if the interceptor failed to cache the condition model before the update
+	cacheError error
+}
```

Comment:
Same as my comment about the patient interceptor -- there are likely some threading issues here...